functions/src/days-before-campaign-ends.ts

```diff
@@ -13,5 +13,12 @@ export const daysBeforeCampaignEnds = (
       1
     );
   }
-  return campaignOptions.endOn - now.getUTCDate() + 1;
+  let possiblyNegativeDaysBeforeCampaignEnds = campaignOptions.endOn - now.getUTCDate() + 1;
+
+  if (possiblyNegativeDaysBeforeCampaignEnds < 0){let lastDayOfMonth = new Date(
+    now.getUTCFullYear(),
+    now.getUTCMonth(),
+    0
+  ).getUTCDate(); return possiblyNegativeDaysBeforeCampaignEnds + lastDayOfMonth;
+  }else return possiblyNegativeDaysBeforeCampaignEnds;
```

Comment:
Formatting looks incorrect. Did prettier produce this?

Reply:
Yes, what's wrong ?

Reply:
Prettier does not produce code like this usually. Let me check.

Reply:
I just checked out the code, applied Prettier and the code looks like this:

```
let possiblyNegativeDaysBeforeCampaignEnds =
    campaignOptions.endOn - now.getUTCDate() + 1;

  if (possiblyNegativeDaysBeforeCampaignEnds < 0) {
    let lastDayOfMonth = new Date(
      now.getUTCFullYear(),
      now.getUTCMonth(),
      0
    ).getUTCDate();
    return possiblyNegativeDaysBeforeCampaignEnds + lastDayOfMonth;
  } else return possiblyNegativeDaysBeforeCampaignEnds;
```

Can you check again?

Reply:
In deed my bad !